app/services/metrc_service/plant/move.rb

```diff
@@ -23,58 +19,74 @@ def transaction
         @transaction ||= get_transaction(:move_batch, @attributes.merge(sub_stage: batch&.zone&.sub_stage&.attributes))
       end
 
-      def prior_move_transactions
-        Transaction.where(
+      def prior_move
+        previous_move = Transaction.where(
           'batch_id = ? AND type = ? AND vendor = ? AND id NOT IN (?)',
           @batch_id,
           :move_batch,
           :metrc,
           transaction.id
-        )
+        ).limit(1).order('created_at desc').first
+
+        return if previous_move.nil?
+
+        batch.completion(previous_move&.completion_id,
+                         include: 'zone,barcodes,sub_zone,action_result,crop_batch_state.seeding_unit')
       end
+      memoize :prior_move
 
       private
 
       def next_step_name
-        transactions = prior_move_transactions
-        return DEFAULT_MOVE_STEP if transactions.count.zero?
-
-        previous_growth_phase = normalized_growth_phase(transactions.last.metadata.dig('sub_stage', 'name'))
+        @completion = batch.completion(@completion_id,
+                                       include: 'zone,barcodes,sub_zone,action_result,crop_batch_state.seeding_unit')
 
-        # Does last move includes new move?
-        is_included = is_included?(previous_growth_phase, normalized_growth_phase)
-        log("Transactions: #{transactions.size}, Previous growth phase: #{previous_growth_phase}, Growth phase is included: #{is_included}, Batch ID #{@batch_id}, completion ID #{@completion_id}")
+        next_step(prior_move, @completion)
+      end
+      memoize :next_step_name
 
-        raise InvalidOperation, "Failed: Substage #{normalized_growth_phase} is not a valid next phase for #{previous_growth_phase}. Batch ID #{@batch_id}, completion ID #{@completion_id}" \
-          unless is_included
+      def next_step(previous_completion = nil, completion = nil) # rubocop:disable Metrics/PerceivedComplexity
+        return DEFAULT_MOVE_STEP if previous_completion.nil? || completion.nil?
 
-        next_step(previous_growth_phase, normalized_growth_phase)
-      end
+        @prior_move ||= previous_completion
```

Comment:
this `@prior_move = ...` will not work with `memoize`. You need to call `prior_move(true)` to refresh the cache, and you cannot assign directly to it.

Reply:
I'm not trying to get a new value, but rather assign a value to the instance var so the `previous_growth_phase` method works as expected.